exporters/otlp/common/src/main/java/io/opentelemetry/exporter/internal/auth/Authenticator.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.exporter.internal.auth;
+
+import io.opentelemetry.exporter.internal.grpc.GrpcExporterBuilder;
+import io.opentelemetry.exporter.internal.okhttp.OkHttpExporterBuilder;
+import java.lang.reflect.Field;
+import java.util.Map;
+import java.util.function.Consumer;
+
+/** Internal class to allow users of OTLP-OkHttp exporters to add support for authentication. */
+public interface Authenticator {
+
+  /**
+   * Method called by the exporter to get headers to be used on a request that requires
+   * authentication.
+   *
+   * @param headers Consumer callback to be used to propagate headers to the underlying OTLP HTTP
+   *     exporter implementation.
+   */
+  void getHeaders(Consumer<Map<String, String>> headers);
+
+  /**
+   * Reflectively access a {@link GrpcExporterBuilder}, or {@link OkHttpExporterBuilder} instance in
+   * field called "delegate" of the instance, and set the {@link RetryPolicy}.
+   *
+   * @throws IllegalArgumentException if the instance does not contain a field called "delegate" of
+   *     a supported type.
+   */
+  public static void setAuthenticatorOnDelegate(Object builder, Authenticator authenticator) {
```

Comment:
```suggestion
 static void setAuthenticatorOnDelegate(Object builder, Authenticator authenticator) {
```
The public keyword is redundant.